src/components/confirm-leave-mixin.cjsx

```diff
@@ -0,0 +1,12 @@
+$ = require 'jquery'
+
+module.exports =
+  # getFlux: -> {store, actions}
+  # getId: -> id 
+  componentDidMount: ->
+    $(window).on("beforeunload", @confirmLeave)
```

Comment:
I've been avoiding it because:

1. we're targeting modern browsers
2. if has a bunch of tasks that perform DOM manipulation which is mostly nullified because of Reactand would lead to questions like "when should I use `$el.append()` or `$el.addClass()`"